Objects/unicodeobject.c

```diff
@@ -15228,13 +15228,16 @@ PyDoc_STRVAR(unicode_doc,
 "str(object='') -> str\n\
 str(bytes_or_buffer[, encoding[, errors]]) -> str\n\
 \n\
-Create a new string object from the given object. If encoding or\n\
-errors is specified, then the object must expose a data buffer\n\
-that will be decoded using the given encoding and error handler.\n\
-Otherwise, returns the result of object.__str__() (if defined)\n\
-or repr(object).\n\
-encoding defaults to sys.getdefaultencoding().\n\
-errors defaults to 'strict'.");
+Create a new string object from the given object.\n\
+\n\
+If a single argument is given, returns the result of object.__str__()\n\
+(if defined) or repr(object).\n\
+\n\
+If encoding or errors or both are specified, then the object must\n\
```

Comment:
Is not "or both" redundant?

Reply:
In english, it's ambiguous, so 'and both' removes that ambiguity, especially as the rest of the paragraph talks about using one option or the other, it might be misunderstood that only one or the other are meant to be used.